src/ray/gcs/gcs_server/gcs_placement_group_scheduler.cc

```diff
@@ -24,12 +24,12 @@ GcsPlacementGroupScheduler::GcsPlacementGroupScheduler(
     boost::asio::io_context &io_context,
     std::shared_ptr<gcs::GcsTableStorage> gcs_table_storage,
     const gcs::GcsNodeManager &gcs_node_manager, GcsResourceManager &gcs_resource_manager,
-    ReserveResourceClientFactoryFn lease_client_factory)
+    std::shared_ptr<rpc::NodeManagerClientPool> raylet_client_pool)
     : return_timer_(io_context),
       gcs_table_storage_(std::move(gcs_table_storage)),
       gcs_node_manager_(gcs_node_manager),
       gcs_resource_manager_(gcs_resource_manager),
-      lease_client_factory_(std::move(lease_client_factory)) {
+      raylet_client_pool_(std::move(raylet_client_pool)) {
```

Comment:
I think we shouldn't use move here because this pool will be shared by multiple places. Thoughts @ffbin ?

Reply:
Yeah, this pool will be shared by multiple places.